src/socket-client.ts

```diff
@@ -241,12 +258,44 @@ export class SocketClient extends EventEmitter {
 
 
         // return success based on connection status
-        return new Promise((resolve, reject) => {
+        await new Promise<void>((resolve, reject) => {
             socket.on("connect_error", () => reject(new Error("[SocketClient] Error connecting")));
             socket.on("connect_timeout", () => reject(new Error("[SocketClient] Error connecting")));
 
+            /**
+             * If the "inner socket handshake" is enabled,
+             * we're expecting a "handshake" event in response to the
+             * initial connection.
+             * 
+             * We will answer this handshake event with our
+             * session parameters that we'd otherwise have sent
+             * through the query parameters.
+             * 
+             * As soon as the handshake is answered, the connection
+             * can be seen as "established"
+             */
+            if (this.socketOptions.enableInnerSocketHandshake) {
+                socket.on("handshake", (cb: Function) => {
```

Comment:
also the name of the event is a bit generic

Reply:
but I guess we can live with it since you added some nice comments. still, having a more descriptive name can be more helpful than having to read comments when checking the code